src/components/stimuli/Stimuli.tsx

```diff
@@ -29,15 +44,80 @@ const Stimuli: React.FC = () => {
       }
     }
 
-    fetchStimuliUrls();
-  }, []);
+    if (!isUploading) fetchStimuliUrls();
+  }, [isUploading]);
+
+  const onFileSelect = (e: any) => {
+    const file: File = e.target.files[0];
+    const formData = new FormData();
+    formData.append('file', file);
+    setUploading(true);
+    fetch(`${process.env.REACT_APP_BACKEND_ADDRESS}/image`, {
+      method: 'POST',
+      body: formData,
+    })
+      .then(res => res.json())
+      .then(() => {
+        setUploading(false);
+        setShowToast(true);
+      });
+  };
+
+  const fileSelector: HTMLInputElement = buildFileSelector();
+  fileSelector.onchange = onFileSelect;
+
+  const handleUploadClick = (e: any) => {
+    e.preventDefault();
+    fileSelector.click();
+  };
 
   return (
     <Container>
+      <Row>
+        <Col>
+          <Button
+            variant="secondary"
+            size="lg"
+            className="my-3"
+            onClick={handleUploadClick}
+            block
+          >
+            {isUploading ? (
+              <Spinner
+                as="span"
+                animation="border"
+                role="status"
+                aria-hidden="true"
+              />
+            ) : (
+              'Upload New Image'
+            )}
+          </Button>
+        </Col>
+      </Row>
       <Row>
         {hasError && <div>Error encountered while loading images.</div>}
         {stimuliUrls && generateCards(stimuliUrls)}
       </Row>
+      {showToast ? (
+        <Row>
+          <Col>
+            <div>
+              <Toast
+                style={{
```

Comment:
With the `absolute` positioning, this is always at the very bottom of all the images, even if the bottom of the image list is off screen. It would be better to somehow position this as a floating element at the bottom of the screen so that it still shows even if the user hasn't scrolled all the way to the bottom